sdk/java-sdk-protobuf-testkit/src/main/java/kalix/javasdk/testkit/EventingTestKit.java

```diff
@@ -208,44 +203,53 @@ interface Topic {
   }
 
   @ApiMayChange
-  interface Message<P> {
-    P getPayload();
+  class MessageBuilder {
+    private final MessageCodec messageCodec;
 
-    Metadata getMetadata();
-
-    /**
-     * Expects message payload to conform to type passed in and returns the typed object if so.
-     * Otherwise, throws an exception.
-     *
-     * @param clazz expected class type for the payload of the message
-     * @return a typed object from the payload
-     * @param <T> the type of the payload
-     */
-    <T> T expectType(Class<T> clazz);
+    public MessageBuilder(MessageCodec messageCodec) {
+      this.messageCodec = messageCodec;
+    }
 
     /**
      * Create a message from a payload plus a subject (that is, the entity key).
      * Automatically adds required default metadata for a CloudEvent.
      *
      * @param payload the message payload
      * @param subject the entity key of which the message is concerned about
-     * @return a Message object to be used in the context of the Testkit
      * @param <T>
+     * @return a Message object to be used in the context of the Testkit
      */
-    static <T> Message<T> of(T payload, String subject) {
-      return new TestKitMessageImpl<>(payload, TestKitMessageImpl.defaultMetadata(payload, subject));
+    public <T> Message<T> of(T payload, String subject) {
+      return new TestKitMessageImpl<>(payload, TestKitMessageImpl.defaultMetadata(payload, subject, messageCodec));
     }
 
     /**
      * Create a message object from a payload plus metadata.
      *
-     * @param payload the message payload
+     * @param payload  the message payload
      * @param metadata the metadata associated with the message
-     * @return a Message object to be used in the context of the Testkit
      * @param <T>
+     * @return a Message object to be used in the context of the Testkit
      */
-    static <T> Message<T> of(T payload, Metadata metadata) {
+    public <T> Message<T> of(T payload, Metadata metadata) {
```

Comment:
Moving these methods here is not backward compatible, not sure how relevant that is.

Reply:
I think it's probably fine since the feature is very recent.

Reply:
Ok, just don't forget to update the webinar (if necessary) :)

Reply:
luckily I did not use the publisher side :D